src/main/java/uk/gov/companieshouse/api/accounts/controller/CurrentPeriodController.java

```diff
@@ -44,36 +52,61 @@ public ResponseEntity create(@Valid @RequestBody CurrentPeriod currentPeriod,
+        CompanyAccountEntity companyAccountEntity = (CompanyAccountEntity) request
+                .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
+
+        String companyAccountId = companyAccountEntity.getId();
+        String requestId = request.getHeader("X-Request-Id");
+
+        ResponseEntity responseEntity;
+        try {
+            ResponseObject<CurrentPeriod> responseObject = currentPeriodService
+                    .create(currentPeriod, transaction, companyAccountId, requestId);
+            responseEntity = apiResponseMapper
+                    .map(responseObject.getStatus(), responseObject.getData(),
+                            responseObject.getValidationErrorData());
+        } catch (DataException ex) {
+            final Map<String, Object> debugMap = new HashMap<>();
+            debugMap.put("transaction_id", transaction.getId());
+            LOGGER.errorRequest(request, ex, debugMap);
+            responseEntity = apiResponseMapper.map(ex);
+        }
+
+        return responseEntity;
     }
 
     @GetMapping
     public ResponseEntity get(HttpServletRequest request) throws NoSuchAlgorithmException {
         LogContext logContext = LogHelper.createNewLogContext(request);
 
+        Transaction transaction = (Transaction) request
+                .getAttribute(AttributeName.TRANSACTION.getValue());
         CompanyAccountEntity companyAccountEntity = (CompanyAccountEntity) request
                 .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
-        if (companyAccountEntity == null) {
 
+        if (companyAccountEntity == null) {
             LOGGER.error("Current Period error: No company account in request session",
                     logContext);
             return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
         }
 
         String companyAccountId = companyAccountEntity.getId();
         String currentPeriodId = currentPeriodService.generateID(companyAccountId);
-        CurrentPeriodEntity currentPeriodEntity = currentPeriodService.findById(currentPeriodId);
-        if (currentPeriodEntity == null) {
+        ResponseObject<CurrentPeriod> responseObject;
 
-            LOGGER.error("Current Period error: No current period found",
-                    logContext);
-            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("transaction_id", transaction.getId());
+
+        try {
+            responseObject = currentPeriodService.findById(currentPeriodId);
+        } catch (DataAccessException dae) {
+            LOGGER.errorRequest(request, dae, debugMap);
+            return apiResponseMapper.map(dae);
         }
 
-        return ResponseEntity.status(HttpStatus.OK).body(currentPeriodEntity);
+        CurrentPeriod currentPeriod = responseObject.getData();
+        return apiResponseMapper.mapGetResponse(currentPeriod, request);
```

Comment:
Since we have the ResponseObject with a ResponseStatus here we should use the regular map method and pass responseObject.getStatus(), getData(), getValidationErrorData()
The map method will need to handle the new ResponseStatus.FOUND/NOT_FOUND values anyway so we should use it when we have a ResponseObject